### PR TITLE
Auth handshake webapp <> api

### PR DESCRIPTION
**Pull request by se7entyse7en**
This PR implements the followings:
- add a redirect from url matching the format of the admin invitation link to the login page,
- add the invitation link to the `appState` of Auth0 to make it available in the callback,
- change the Auth0 callback handler to always redirect to `/callback`,
- call `/v1/accept` endpoint of api if the invitation link correspond to an admin link,
- redirect to waiting page on succes.

### CODE REVIEW

**Comment by se7entyse7en** [comment]
To test this locally, I've used [this version of the api](https://github.com/athenianco/athenian-api/pull/75).

File: src/js/pages/auth/Callback.jsx [hunk 1]
```diff
@@ -1,10 +1,55 @@
...
+        if (inviteLink) {
+          const body = new InvitationLink(inviteLink);
+          api.checkInvitation(body)
+            .then(check => {
+              if (check.valid && check.active) {
+                const invType = check.type;
+                return api.acceptInvitation(body);
+              }
+
+              return Promise.reject(new Error('invalid invitation'));
```
Comment (dpordomingo) [inline src/js/pages/auth/Callback.jsx]: I tried many times, and in different situations, I got different errors without being explained in the UI.

Example, being logged out, I just called `invite/accept` with a just generated invitation, and got:
```json
{"type": "/errors/LockedError",
  "title": "Locked",
  "status": 423,
  "detail": "Invitation was not found."
}
```
But the error details were not exposed in the UI, and got a forever-waiting page.
image

**Comment by se7entyse7en** [comment]
Let me know if you have other problems in making the api work with this PR.

**Comment by dpordomingo** [comment]
I finally was able to reproduce this new feature with the `latest` api docker image.
I missed error messages when debugging, so I added them on my own.
PR https://github.com/athenianco/athenian-webapp/pull/58 addresses my major concerns about this PR, so please consider adding it.

File: src/js/App.jsx [hunk 2]
```diff
@@ -8,14 +8,16 @@ import { Auth0Provider } from 'js/services/react-auth0-spa';
 import history from 'js/services/history';
 import Development from 'js/components/development';
 
+import {
+  Redirect,
```
[the remark this answers is not in the document]
Comment (dpordomingo) [reply src/js/App.jsx]: addressed by #58

File: src/js/pages/auth/Callback.jsx (hunk 1 above)
Comment (dpordomingo) [reply src/js/pages/auth/Callback.jsx]: addressed by #58

**Pull request merged**